DXMainClient/Domain/Multiplayer/Map.cs

```diff
@@ -466,33 +465,12 @@ public Texture2D LoadPreviewTexture()
             if (!Official)
             {
                 // Extract preview from the map itself
+                System.Drawing.Bitmap preview = MapPreviewExtractor.ExtractMapPreview(mapIni);
 
-                if (mapIni.GetSectionKeys("PreviewPack").Count == 0)
-                {
-                    Logger.Log(mapIni.FileName + " - no [PreviewPack] exists");
-                    return AssetLoader.CreateTexture(Color.Black, 10, 10);
-                }
-                    
-                if (mapIni.GetStringValue("PreviewPack", "1", string.Empty) ==
-                    "yAsAIAXQ5PDQ5PDQ6JQATAEE6PDQ4PDI4JgBTAFEAkgAJyAATAG0AydEAEABpAJIA0wBVA")
-                {
-                    Logger.Log(mapIni.FileName + " - Hidden preview detected - not extracting.");
+                if (preview != null)
+                    return AssetLoader.TextureFromImage(preview);
```

Comment:
`AssetLoader.TextureFromImage` will return null on failure, should handle that case and create the dummy 10x10 black texture. Although the existing checks seem to make sure that the extracted bitmap is correct, I don't want to take chances if some kind of a corner-case is found where the bitmap -> Texture2D conversion fails and someone could then crash the client through that.